CMakeLists.txt

```diff
@@ -114,127 +116,57 @@ if (NOT DEFINED VERONA_LLVM_LOCATION)
     USES_TERMINAL_CONFIGURE true
   )
 
-  if (DEFINED VERONA_EXTRA_CMAKE_ARGS)
-    message (WARNING "Do not use VERONA_EXTRA_CMAKE_ARGS! Value ignored")
-  endif ()
-  set (VERONA_EXTRA_CMAKE_ARGS)
-
-  # Point Verona at the MLIR
+  # Point Verona at the MLIR install, we just built/downloaded
   list (APPEND VERONA_EXTRA_CMAKE_ARGS
-    -DVERONA_LLVM_LOCATION=${MLIR_INSTALL}/install
+    -DVERONA_LLVM_LOCATION=${CMAKE_BINARY_DIR}/$<CONFIG>/mlir/install
   )
-
-  # Define project level defaults
-  list (APPEND VERONA_EXTRA_CMAKE_ARGS
-    -DCMAKE_CXX_STANDARD=17)
-
-  # Use top-level install directory when building
-  # as a subproject.
-  list (APPEND VERONA_EXTRA_CMAKE_ARGS
-    -DCMAKE_INSTALL_PREFIX=${VERONA_LOCAL_DIST})
-  list (APPEND VERONA_DONT_PASS CMAKE_INSTALL_PREFIX)
-
-  get_cmake_property(CACHE_VARS CACHE_VARIABLES)
-  foreach (CACHE_VAR ${CACHE_VARS})
-    list (FIND VERONA_DONT_PASS ${CACHE_VAR} INDEX)
-    if (${INDEX} EQUAL -1)
-      list(APPEND VERONA_EXTRA_CMAKE_ARGS
-        -D${CACHE_VAR}=${${CACHE_VAR}}
-      )
-    endif()
-  endforeach ()
-
-
-  ExternalProject_Add(verona
-    SOURCE_DIR ${CMAKE_SOURCE_DIR}
-    DEPENDS external
-    BUILD_ALWAYS true
-    USES_TERMINAL_BUILD true
-    USES_TERMINAL_CONFIGURE true
-    USES_TERMINAL_INSTALL true
-    USES_TERMINAL_TEST true
-    CMAKE_ARGS ${VERONA_EXTRA_CMAKE_ARGS})
-
-  enable_testing()
-
-  add_subdirectory(testsuite)
-
-  # Main target does not test runtime, build subdirectory for that
-  if (RT_TESTS)
-    add_subdirectory(src/rt)
-  endif()
-
-  include(ProcessorCount)
-  ProcessorCount(N)
-
-  # Adds a target check that runs the tests.
-  add_custom_target(check
-    COMMAND ${CMAKE_CTEST_COMMAND} -C $<CONFIG> -j ${N} --output-on-failure --timeout 400 --interactive-debug-mode 0
-    USES_TERMINAL
-  )
-
-  # If RT_TESTS are enabled, we need to make sure they are
-  # up-to-date before we run tests.
-  if (RT_TESTS)
-    add_dependencies(check rt_tests)
```

Comment:
This line is missing in the new version I think.

Reply:
Good catch got dropped in a merge conflict when I rebased.